mgmt/rest/client/client_func_test.go

```diff
@@ -160,332 +164,341 @@ func TestSnapClient(t *testing.T) {
-					<-wait
-					a.Lock()
-					So(len(a.events), ShouldEqual, 5)
-					a.Unlock()
-					So(a.events[0], ShouldEqual, "task-started")
-					for x := 2; x <= 4; x++ {
-						So(a.events[x], ShouldEqual, "metric-event")
-					}
+					})
 				})
 			})
 		})
-	})
-	Convey("UnloadPlugin", t, func() {
-		Convey("unload unknown plugin", func() {
-			p := c.UnloadPlugin("not a type", "foo", 3)
-			So(p.Err, ShouldNotBeNil)
-			So(p.Err.Error(), ShouldEqual, "plugin not found")
-		})
-		Convey("unload one of multiple", func() {
-			p1 := c.GetPlugins(false)
-			So(p1.Err, ShouldBeNil)
-			So(len(p1.LoadedPlugins), ShouldEqual, 3)
-
-			p2 := c.UnloadPlugin("collector", "mock", 2)
-			So(p2.Err, ShouldBeNil)
-			So(p2.Name, ShouldEqual, "mock")
-			So(p2.Version, ShouldEqual, 2)
-			So(p2.Type, ShouldEqual, "collector")
-
-			p3 := c.UnloadPlugin("publisher", "file", 3)
-			So(p3.Err, ShouldBeNil)
-			So(p3.Name, ShouldEqual, "file")
-			So(p3.Version, ShouldEqual, 3)
-			So(p3.Type, ShouldEqual, "publisher")
-		})
-		Convey("unload when only one plugin loaded", func() {
-			p1 := c.GetPlugins(false)
-			So(p1.Err, ShouldBeNil)
-			So(len(p1.LoadedPlugins), ShouldEqual, 1)
-			So(p1.LoadedPlugins[0].Name, ShouldEqual, "mock")
-
-			p2 := c.UnloadPlugin("collector", "mock", 1)
-			So(p2.Err, ShouldBeNil)
-			So(p2.Name, ShouldEqual, "mock")
-			So(p2.Version, ShouldEqual, 1)
-			So(p2.Type, ShouldEqual, "collector")
-
-			p3 := c.GetPlugins(false)
-			So(p3.Err, ShouldBeNil)
-			So(len(p3.LoadedPlugins), ShouldEqual, 0)
+		Convey("UnloadPlugin", func() {
+			Convey("unload unknown plugin", func() {
+				p := c.UnloadPlugin("not a type", "foo", 3)
+				So(p.Err, ShouldNotBeNil)
+				So(p.Err.Error(), ShouldEqual, "plugin not found")
+			})
+			Convey("unload one of multiple", func() {
+				p1 := c.GetPlugins(false)
+				So(p1.Err, ShouldBeNil)
+				So(len(p1.LoadedPlugins), ShouldEqual, 3)
+
+				p2 := c.UnloadPlugin("collector", "mock", 2)
+				So(p2.Err, ShouldBeNil)
+				So(p2.Name, ShouldEqual, "mock")
+				So(p2.Version, ShouldEqual, 2)
+				So(p2.Type, ShouldEqual, "collector")
+
+				p3 := c.UnloadPlugin("publisher", "file", 3)
+				So(p3.Err, ShouldBeNil)
+				So(p3.Name, ShouldEqual, "file")
+				So(p3.Version, ShouldEqual, 3)
+				So(p3.Type, ShouldEqual, "publisher")
+			})
+			Convey("unload when only one plugin loaded", func() {
+				p1 := c.GetPlugins(false)
+				So(p1.Err, ShouldBeNil)
+				So(len(p1.LoadedPlugins), ShouldEqual, 1)
+				So(p1.LoadedPlugins[0].Name, ShouldEqual, "mock")
+
+				p2 := c.UnloadPlugin("collector", "mock", 1)
+				So(p2.Err, ShouldBeNil)
+				So(p2.Name, ShouldEqual, "mock")
+				So(p2.Version, ShouldEqual, 1)
+				So(p2.Type, ShouldEqual, "collector")
+
+				p3 := c.GetPlugins(false)
+				So(p3.Err, ShouldBeNil)
+				So(len(p3.LoadedPlugins), ShouldEqual, 0)
+			})
 		})
 	})
-
-	c = New("http://localhost:127.0.0.1:-1", "v1", true)
+	c, err := New("http://localhost:127.0.0.1:-1", "v1", true)
```

Comment:
I think this brings up an issue with the change in that `http://localhost:127.0.0.1:-1` is an invalid url and should not return a client, but url.Parse is setting `localhost:127.0.0.1:-1` as the host and a client is being returned.